src/libraries/System.Private.CoreLib/src/System/Buffers/ConfigurableArrayPool.cs

```diff
@@ -258,6 +268,12 @@ internal void Return(T[] array)
                 {
                     if (lockTaken) _lock.Exit(false);
                 }
+
+                ArrayPoolEventSource log = ArrayPoolEventSource.Log;
+                if (log.IsEnabled() && !returned)
+                {
+                    log.BufferDropped(array.GetHashCode(), _bufferLength, _poolId, Id, ArrayPoolEventSource.BufferDroppedReason.Full);
```

Comment:
is this cheaper? presumably buffers are usually returned, why access and test Log in that case.
```suggestion
                if (!returned)
                {
                    ArrayPoolEventSource log = ArrayPoolEventSource.Log;
                    if  (log.IsEnabled())
                    {
                        log.BufferDropped(array.GetHashCode(), _bufferLength, _poolId, Id, ArrayPoolEventSource.BufferDroppedReason.Full);
```